### PR TITLE
Fix ternary operator handling to support jinja2 2.10.x and sanity fixes

### PR DESCRIPTION
**Pull request by NilashishC**
Depends-On:  https://github.com/ansible/project-config/pull/762

### CODE REVIEW

**Comment by GomathiselviS** [comment]
recheck

**Comment by NilashishC** [comment]
recheck

File: plugins/module_utils/network/nxos/argspec/bgp_global/bgp_global.py [hunk 1]
```diff
@@ -204,9 +204,10 @@ class Bgp_globalArgs(object):  # pylint: disable=R0903
                         },
                         "password": {
                             "type": "dict",
+                            "no_log": False,
```
Comment (GomathiselviS) [inline plugins/module_utils/network/nxos/argspec/bgp_global/bgp_global.py:207]: I think no_log = True will make sense here.

**Review by GomathiselviS — CHANGES_REQUESTED** [review]
no_log should be set to True for keys with sensitive information.

File: plugins/module_utils/network/nxos/argspec/bgp_global/bgp_global.py (hunk 1 above)
Comment (NilashishC) [reply plugins/module_utils/network/nxos/argspec/bgp_global/bgp_global.py:207]: I remember when we discussed this the last time, we concluded that what is sensitive and what is not is subjective to the end user. And as such, a toggle that let's the user sacrifice round trip for security would come in handy. For modules that are already released, adding no_log: True would be a breaking change. I think we need to discuss this again. Will keep the PR open till then.

**Review by ashwini-mhatre — APPROVED** [review]

**Comment by NilashishC** [comment]
> no_log should be set to True for keys with sensitive information.

Resolved.

**Review by GomathiselviS — APPROVED** [review]

**Comment by NilashishC** [comment]
recheck

**Pull request merged**